graphql_compiler/compiler/emit_sql.py

```diff
@@ -96,6 +98,17 @@ def _find_columns_used_outside_folds(sql_schema_info, ir):
             if child_location_info.recursive_scopes_depth > location_info.recursive_scopes_depth:
                 used_columns.setdefault(child_location.query_path, set()).add(edge.from_column)
 
+    # Columns used in the base case of CTE recursions should be made available from parent scope
+    for location, _ in ir.query_metadata_table.registered_locations:
+        if isinstance(location, FoldScopeLocation):
+            continue
+        for recurse_info in ir.query_metadata_table.get_recurse_infos(location):
+            traversal = "{}_{}".format(recurse_info.edge_direction, recurse_info.edge_name)
+            used_columns[location.query_path] = used_columns.get(location.query_path, set()).union(
+                used_columns[location.query_path + (traversal,)]
+            )
+            used_columns.setdefault(location.query_path, set()).add(edge.from_column)
```

Comment:
I'm confused by these two consecutive statements. A few things that are confusing:
- `used_columns` has no type hint, so I don't know what to expect inside it (the function itself is also missing type hints)
- I'm not sure why both statements are assigning to what appears to be the same entry in the dict, `used_columns[location.query_path]`.
- Furthermore, I'm confused why the second statement uses `setdefault()` when the first one guarantees that the value at that key is already present and already a set.